src/onion_messenger.rs

```diff
@@ -699,6 +702,32 @@ pub(crate) async fn relay_outgoing_msg_event(
     }
 }
 
+// A simple offers message handler.
+#[derive(Clone)]
+pub struct InvoiceHandler {
+    pub invoices: Arc<Mutex<VecDeque<Bolt12Invoice>>>,
+}
+
+impl OffersMessageHandler for InvoiceHandler {
+    fn handle_message(&self, message: OffersMessage) -> Option<OffersMessage> {
+        match message {
+            OffersMessage::Invoice(invoice) => {
+                println!("Received an invoice offers message.");
+                self.invoices.lock().unwrap().push_back(invoice);
```

Comment:
Shouldn't we track that we actually sent an invoice request for the offer, and discard the invoice if not?